src/main/java/com/google/googleidentity/security/UserSession.java

```diff
@@ -0,0 +1,43 @@
+package com.google.googleidentity.security;
+
+import com.google.googleidentity.user.UserDetails;
+import com.google.inject.servlet.SessionScoped;
+
+import java.io.Serializable;
+
+@SessionScoped
+public class UserSession implements Serializable {
+
+    private static final long serialVersionUID = 3L;
+
+    private UserDetails user = null;
+
+    private String olduri = "";
```

Comment:
This seems to have similar semantics that would benefit from java.util.Optional instead of setting to the empty string.